### PR TITLE
begin ping plugin

### PR DESCRIPTION
**Pull request by AmarOk1412**
**DO NOT MERGE**

cf #37 

# TODO

- [ ] wait v24 (https://zestedesavoir.com)
- [ ] Use jest for tests.
- Configurable?

### CODE REVIEW

File: packages/remark-ping-zds/src/index.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    return value.indexOf('@', keep.index)
+  }
+  return -1
+}
+
+function plugin () {
+  function zdsMemberExists (username) {
+    //TODO https://github.com/zestedesavoir/zds-site/issues/4382
+    let isMember = false
+    const res = request('GET', `http://zestedesavoir.com/api/membres/?search=${username}`)
```
Comment (artragis) [inline packages/remark-ping-zds/src/index.js]: il faudrait que ça soit configurable, non?

File: packages/remark-ping-zds/src/index.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  }
+  inlineTokenizer.locator = locator
+
+  const Parser = this.Parser
+
+  // Inject inlineTokenizer
+  const inlineTokenizers = Parser.prototype.inlineTokenizers
+  const inlineMethods = Parser.prototype.inlineMethods
+  inlineTokenizers.zping = inlineTokenizer
+  inlineMethods.splice(inlineMethods.indexOf('text'), 0, 'zping')
```
Comment (artragis) [inline packages/remark-ping-zds/src/index.js]: I think we should remove the `z` part for all `remarl_xxx` plugins.

File: packages/remark-ping-zds/src/index.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      return eat(total)({
+        type: 'ping',
+        children: [{
+          type: 'text',
+          value: username
+        }],
+        data: {
+          hName: 'a',
+          hProperties: {
+            href: `/membres/voir/${username}/`,
```
Comment (artragis) [inline packages/remark-ping-zds/src/index.js]: Should be configurable.

File: packages/remark-ping-zds/src/index.js (hunk 1 above)
Comment (AmarOk1412) [inline packages/remark-ping-zds/src/index.js]: Hmmm, I'm not sure actually. It's a very specific plugin for a specific case. If anyone want to use this plugin for another website, they must provide an endpoint which return a code 200 if the user exists.

**Comment by vhf** [comment]
Thanks! That was the last important step!

**Pull request closed without merging**